clusterloader2/pkg/measurement/common/slos/windows_node_memory_usage.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package slos
+
+import (
+	"math"
+	"time"
+
+	"github.com/prometheus/common/model"
+	"k8s.io/klog"
+	"k8s.io/perf-tests/clusterloader2/pkg/measurement"
+	measurementutil "k8s.io/perf-tests/clusterloader2/pkg/measurement/util"
+	"k8s.io/perf-tests/clusterloader2/pkg/util"
+)
+
+const (
```

Comment:
There is a lot of boilerplate and copy-pasting here. Can we either:
1. Keep the windows_node_resource_usage measurement, that will do both cpu and memory
2. Extract the common parts somewhere and keep the different instance as short and simple as possible? E.g. you could have an abstract "WindowsResourceUsage" measurement that would take care of all the boilerplate. Then the particular instances of this measurement (e.g. cpu, memory, etc) would only have to provide the differentiating parts, e.g. the query to execute, name of fields in the summary, etc. 

WDYT?

Reply:
+1

Reply:
Thanks, make sense. 
For #1, as all the prometheus measurements need to implement the Gatherer interface in prometheus_measurement.go, and the `Execute()` method in prometheus_measurement.go could return multiple summary, but the Gather() method return only one summary, which seems not reasonable. Could we change the Gather() method to return list of summary? That will change the existing APIResponsiveness & NetworkingProgramming prometheus measurements also.
I'd prefer this modification, as it enables us to return multiple promethues measurement summary in one file to avoid the abstract layer. But if you think it might break the existing measurements, I will go for #2

Reply:
Yeah, that's a great observation, the Gather in prometheus_measurement should return []measurement.Summary.
Could you send a separate PR to fix that?

Reply:
Sure

Reply:
Created a PR: https://github.com/kubernetes/perf-tests/pull/1159

Reply:
I already approved that one. Once that merges, please rebase this PR.

Reply:
rebased the PR, please take a look again